File: adapterx/src/main/java/com/dhy/adapterx/LayoutName.java
package com.dhy.adapterx;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.PARAMETER;
import static java.lang.annotation.ElementType.TYPE;

@Retention(RetentionPolicy.RUNTIME)
@Target({TYPE, PARAMETER})
public @interface LayoutName {
    String value();
}

File: lib/src/main/java/com/dhy/adapterx/LayoutName.kt
package com.dhy.adapterx

/**
 * use {@link IViewHolder} then set value to parameter.
 * eg: IViewHolder(v, R.layout.item)
 */
@Deprecated("use IViewHolder")
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class LayoutName(val value: String)